src/Lykke.Logs/AzureTablePersistence/AzureTableLogPersistenceQueue.cs

```diff
@@ -0,0 +1,155 @@
+            }
+
+            _storage = storage ?? throw new ArgumentNullException(nameof(storage));
+            _rowKeyGenerator = rowKeyGenerator ?? throw new ArgumentNullException(nameof(rowKeyGenerator));
+            _batchSizeThreshold = batchSizeThreshold;
+            _degreeOfPersistenceParallelism = degreeOfPersistenceParallelism;
+            _maxBatchLifetime = maxBatchLifetime ?? TimeSpan.FromSeconds(5);
+
+            lastResortLogFactory = lastResortLogFactory ?? LastResortLogFactory.Instance;
+
+            _log = lastResortLogFactory.CreateLog(this);
+
+            _batchBlock = new BatchBlock<TLogEntity>(batchSizeThreshold);
+            _groupBatchBlock = new TransformManyBlock<TLogEntity[], IGrouping<string, TLogEntity>>(batch => GroupEntriesBatch(batch));
+            _persistGroupBlock = new ActionBlock<IGrouping<string, TLogEntity>>(
+                // ReSharper disable once ConvertClosureToMethodGroup
+                group => PersistEntriesGroup(group),
+                new ExecutionDataflowBlockOptions
+                {
+                    MaxDegreeOfParallelism = degreeOfPersistenceParallelism
+                });
+
+            _batchBlock.LinkTo(_groupBatchBlock);
+            _groupBatchBlock.LinkTo(_persistGroupBlock);
+
+            ExtendBatchExpiration();
+
+            _timer = new TimerTrigger(logName, TimeSpan.FromMilliseconds(50), lastResortLogFactory)
+                .DisableTelemetry();
+            _timer.Triggered += HandleTimerTriggered;
+            _timer.Start();
+        }
+
+        /// <inheritdoc />
+        public void Enqueue(TLogEntity entry)
+        {
+            _batchBlock.Post(entry);
+        }
+
+        public void Dispose()
+        {
+            _batchBlock.Complete();
+            _batchBlock.Completion.ConfigureAwait(false).GetAwaiter().GetResult();
+            
+            _groupBatchBlock.Complete();
+            _groupBatchBlock.Completion.ConfigureAwait(false).GetAwaiter().GetResult();
+
+            _persistGroupBlock.Complete();
+            _persistGroupBlock.Completion.ConfigureAwait(false).GetAwaiter().GetResult();
+
+            _timer.Dispose();
+        }
+
+        private IEnumerable<IGrouping<string, TLogEntity>> GroupEntriesBatch(TLogEntity[] batch)
+        {
+            ExtendBatchExpiration();
+            
+            return batch.GroupBy(e => e.PartitionKey);
+        }
+
+        private Task PersistEntriesGroup(IGrouping<string, TLogEntity> group)
+        {
+            return _storage.InsertBatchAndGenerateRowKeyAsync(
+                group.ToArray(),
+                (entity, retryNum, batchItemNum) => _rowKeyGenerator.Generate(entity, retryNum, batchItemNum));
+        }
+
+        private Task HandleTimerTriggered(ITimerTrigger timer, TimerTriggeredHandlerArgs args, CancellationToken cancellationtoken)
+        {
+            if (DateTime.UtcNow > _currentBatchExpirationMoment)
+            {
+                _batchBlock.TriggerBatch();
```

Comment:
Documentation tells nothing about possible exceptions. I hope this will not throws anything, but even if it could, then TimerTrigger will handle and log it.